docs/src/manual/constraints.md

```diff
@@ -55,4 +56,23 @@ apply!(K, f, ch)
 a = K\f
 apply!(a, ch) # enforces affine constraints
 
-```
\ No newline at end of file
+```
+
+Note: Replacing `f` with `r=-f` and solving `a = -K\r` in the example above may give the wrong result.
+Instead, use the procedure for nonlinear problems when applying affine constraints to such problem formulations.
+
+### Solving nonlinear problems
+A Newton-type solution method for the nonlinear system `r(a)=0` uses the 
+update `Δa=-K\r` (where `K=∂r/∂a` if the standard Newton's method is used).
+```julia
+a = initial_guess(...)  # Make any initial guess for a here, e.g. `a=zeros(ndofs(dh))`
+apply!(a, ch)           # Make the guess fulfill all constraints in `ch`
+for iter in 1:maxiter
+    doassemble!(K, r, ...)
+    apply_zero!(K, r, ch)   # Modify `K` and `r` to account for the constraints. 
+    check_convergence(r, ...) && break # Only check convergence after `apply_zero!(K, r, ch)`
+    Δa=-K\r                 # Calculate update
```

Comment:
```suggestion
    Δa = K \ r             # Calculate update
```

Reply:
From a performance point of view to avoid allocations?
It seems a bit counter-intuitive to me to have `Δa` imply the negative update.

Reply:
Yea, avoids allocation of `K \r ` (`tmp = K \ r; Δa = -tmp`).

>  It seems a bit counter-intuitive to me to have Δa imply the negative update.

I guess it depends on how you state the problem, this is how it is put up in e.g. https://ferrite-fem.github.io/Ferrite.jl/stable/examples/hyperelasticity/#Newton's-Method

Reply:
Fair point, let's just go with the negative update - it is clear in any case!
I'll update https://ferrite-fem.github.io/Ferrite.jl/stable/reference/boundary_conditions/#Ferrite.apply_zero! to be consistent also then.